File: src/main/java/com/shipout/controller/UserController.java
package com.shipout.controller;

import com.shipout.base.R;
import com.shipout.entity.User;
import com.shipout.service.UserService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/user")
public class UserController {


    @Autowired
    private UserService userService;


    @GetMapping("/list")
    public R<List<User>> getUserList() {

        List<User> userList = userService.getUserList();

        return R.ok(userList);

    }


    @GetMapping("/getById")
    public R<User> getUserById(Integer id) {

        User user = userService.findUserById(id);

        return R.ok(user);

    }
    @PostMapping("/addUser")
    R<User> getUserById(@RequestBody User user) {
        return R.ok(null);
    }

}

File: src/main/java/com/shipout/controller/UserController.kt
package com.shipout.controller

import com.shipout.base.*
import com.shipout.entity.*
import com.shipout.service.*
import org.springframework.beans.factory.annotation.*
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/user")
class UserController {
    @Autowired
    private lateinit var userService: UserService

    @GetMapping("/list")
    fun getUserList(): R<List<User>?> {
        val userList = userService.getUserList()
        return R.Companion.ok<List<User>?>(userList)
    }

    @GetMapping("/getById")
    fun getUserById(id: Int): R<User?> {
        val user = userService!!.findUserById(id)
        return R.Companion.ok<User?>(user)
    }

    @PostMapping("/addUser")
    fun addUser(@RequestBody user: User?): R<User?> {
        return R.Companion.ok<User?>(null)
    }
}
